x/delegation/types/genesis.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	assetstypes "github.com/ExocoreNetwork/exocore/x/assets/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewGenesis returns a new genesis state with the given inputs.
+func NewGenesis(
+	delegations []DelegationsByStaker,
+) *GenesisState {
+	return &GenesisState{
+		Delegations: delegations,
+	}
+}
+
+// DefaultGenesis returns the default genesis state
+func DefaultGenesis() *GenesisState {
+	return NewGenesis([]DelegationsByStaker{})
+}
+
+// Validate performs basic genesis state validation returning an error upon any
+// failure.
+func (gs GenesisState) Validate() error {
+	stakers := make(map[string]struct{}, len(gs.Delegations))
```

Comment:
During the verification process, for each level of data (stakeholder, asset, operator), a separate hash table is used to check for uniqueness. Be aware that memory usage may increase when processing large amounts of data.

Reply:
As discussed, added a TODO for this.